py4web/core.py

```diff
@@ -1710,7 +1711,15 @@ def start_server(kwargs):
 
     # Catch interrupts like Ctrl-C if needed
     def kill_all(sig, _):
-        os.kill(os.getpid(), signal.SIGKILL)
+        # better way to kill, adapted from: https://stackoverflow.com/a/70565806
+        main_process = psutil.Process()
+        children = main_process.children(recursive=True)
+        for child in children:
+            child.terminate()  # friendly termination
+        _, still_alive = psutil.wait_procs(children, timeout=2)
+        for child in still_alive:
+            child.kill()  # unfriendly termination
+        main_process.terminate()
```

Comment:
does this really work? The issue is with threads that seem to get stuck, not processes?

Reply:
Well, worst case its the same as the previous SIGKILL, but also works on windows.

I haven't noticed any threads sticking around, but i'm also on Windows so its only tested there.

Anyone working on linux could always just test it with `pip uninstall py4web && pip install git+https://github.com/laundmo/py4web`

Reply:
> The issue is with threads that seem to get stuck, not processes?

After further consideration, I think I may have misunderstood originally. You're questioning whether this will stop even if a thread is currently "stuck" doing some long-term work. As a test, I added a `while True:` loop to my index controller, which of course leads to that page just waiting. I can still stop py4web with `CTRL+C` even after this, so it seems that isn't a concern.

Reply:
I will test this some more. if it works on linux then I trust you it works on windows. ;-)

Reply:
@mdipierro Any update on this? would be nice to switch back from using my fork to the main release

Reply:
I tested it on Linux and this fails to stop the threads. Can you provide a PR that only affects windows?

Reply:
Could you please elaborate on the exact difference you observed between this PR and the released version which just sends SIGKILL?

Threads can't exist without a process on any OS, so does it actually fail to kill some process?